src/sage/rings/polynomial/multi_polynomial_sequence.py

```diff
@@ -908,15 +908,10 @@ def connection_graph(self):
             sage: F.connection_graph()
             Graph on 3 vertices
         """
-        V = sorted(self.variables())
         from sage.graphs.graph import Graph
         g = Graph()
-        g.add_vertices(sorted(V))
         for f in self:
-            v = f.variables()
```

Comment:
Is this really equivalent? In the older version it seems that added edges were `(x0, x1)`, `(x0, x2)`, `(x0, x3)`, ... etc whereas with `.add_path` you would end up with `(x0, x1)`, `(x1, x2)`, `(x2, x3)`, ... Maybe it does not matter.

Reply:
Given the specifications it should rather be a clique instead of a star or a path (ie both versions are not following the specifications). For example the following is equally wrong in the old and new versions
```python
sage: B.<x,y,z> = BooleanPolynomialRing()
sage: F = Sequence([x + y + z])
sage: F.connection_graph().num_edges()
2
```
Though, when you compute connected components it of course does not matter. I think this should be clarified.

Reply:
A solution could be to add parameter `clique=True` to `connection_graph` with default behavior to add cliques and otherwise to add paths (less edges, so more efficient for connected components).

Reply:
Note that you can rewrite method `connected_components` without building the graph. You just need a union-find data structure, so a `DisjointSet` of the variables.

```py
def connected_components(self):
    """
    """
    # Use a union-find data structure to encode relationships between variables,
    # i.e., that they belong to a same connected component
    DS = DisjointSet(self.variables())
    for f in self:
        var = f.variables()
        for v in var[1:]:
            DS.union(var[0], v)
    # Get one representative element per set of variables
    roots = list(D.root_to_elements_dict().keys())
    
    P = [[] for _ in range(len(roots))]
    for f in self:
        u = f.variables()[0]
        for i, r in enumerate(roots):
            if D.find(u) == D.find(r):
                P[i].append(f)
                break
    P = sorted([PolynomialSequence(sorted(p)) for p in P])
    return P
```

Reply:
> Given the specifications it should rather be a clique instead of a star or a path (ie both versions are not following the specifications).

That was my first thought, but since (a) the original code didn't do that (b) adding a clique is a bit slower (c) adding a clique was broken before #35511, I decided on this. In fact, the example in this method is what led me to spot #35511.

> Though, when you compute connected components it of course does not matter. I think this should be clarified.

If we can have an independent implementation of connected components then this doesn't matter.

> A solution could be to add parameter `clique=True` to `connection_graph` with default behavior to add cliques and otherwise to add paths (less edges, so more efficient for connected components).

I like this, will do, and make connected_components use the `clique=False` option for speed.

Reply:
> Note that you can rewrite method `connected_components` without building the graph. You just need a union-find data structure, so a `DisjointSet` of the variables.
> 
> ```python
> def connected_components(self):
>     """
>     """
>     # Use a union-find data structure to encode relationships between variables,
>     # i.e., that they belong to a same connected component
>     DS = DisjointSet(self.variables())
>     for f in self:
>         var = f.variables()
>         for v in var[1:]:
>             DS.union(var[0], v)
>     # Get one representative element per set of variables
>     roots = list(D.root_to_elements_dict().keys())
>     
>     P = [[] for _ in range(len(roots))]
>     for f in self:
>         u = f.variables()[0]
>         for i, r in enumerate(roots):
>             if D.find(u) == D.find(r):
>                 P[i].append(f)
>                 break
>     P = sorted([PolynomialSequence(sorted(p)) for p in P])
>     return P
> ```

It might be a good idea to do an implementation like this. Bear in mind that here you are calling `f.variables()` several times and that is a slow part (really bad before #35510).

I think it's better if you do this in a follow up PR (I can review there).

Reply:
> That was my first thought, but since (a) the original code didn't do that
The original code adds a clique by adding a set of stars

> It might be a good idea to do an implementation like this. Bear in mind that here you are calling f.variables() several times and that is a slow part (really bad before https://github.com/sagemath/sage/pull/35510).
Will do. It's easy to avoid calling twice `f.variables()`

Reply:
> > That was my first thought, but since (a) the original code didn't do that
> 
> The original code adds a clique by adding a set of stars

No, the original code added just one star centered in the first vertex. And the only example gives the same graph with either star, path, or clique.